File: codegen/results/v002/ollama/qwen3.5_4b-nvfp4_nothink/20260412_082424/python/001_csv_to_json/solution.py
import json
import csv

with open('input/data.csv', 'r', newline='', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    rows = [
        {"Name": row["Name"], "Age": int(row["Age"]), "Email": row["Email"], "Score": float(row["Score"])}
        for row in reader
    ]

print(json.dumps(rows))

File: codegen/results/v002/ollama/qwen3.5_4b-nvfp4_nothink/20260412_082424/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';

const filePath = process.argv[2];
if (!filePath) {
  console.error('Usage: tsx script.ts input/data.csv');
  process.exit(1);
}

const content = fs.readFileSync(filePath, 'utf-8');
const lines = content.split('\n').slice(1); // skip header

const result: any[] = [];
for (const line of lines) {
  if (!line.trim()) continue;
  
  const parts = line.split(',');
  const name = parts[0].trim();
  const age = parseInt(parts[1], 10);
  const email = parts[2].trim();
  const score = parseFloat(parts[3]);

  result.push({ Name: name, Age: age, Email: email, Score: score });
}

console.log(JSON.stringify(result));